ambry-api/src/main/java/com.github.ambry/router/ReadableStreamChannel.java

```diff
@@ -45,4 +46,33 @@
    * @return the {@link Future} that will eventually contain the result of the operation.
    */
   public Future<Long> readInto(AsyncWritableChannel asyncWritableChannel, Callback<Long> callback);
+
+  /**
+   * Set the digest algorithm to use on the data that is being streamed from the channel. Once the channel is emptied,
+   * the digest can be obtained via {@link #getDigest()}.
+   * <p/>
+   * This function is ideally called before {@link #readInto(AsyncWritableChannel, Callback)}. After a call to
```

Comment:
ideally - is there a scenario where this setting the algorithm after the `readInto()` makes sense? I'd say rather than allowing it for implementations, just disallow it at the interface level, unless you think there is a value. Just keeps it tight.